### PR TITLE
Update gtfs_headway_analysis.py

### PR DESCRIPTION
**Pull request by carlhiggs**
Updated to account for frequencies.txt, and use folder of assets with process/data/GTFS (ie. unified data folder with rest of project)   --- that latter change may not be desireable, but the other one is important.  Also includes fixes with taking mean across dates, rather than minimum; and creating a 'direction_id' field, and ensuring this does not have nulls, to avoid if else conditions related to this.

### CODE REVIEW

File: process/GTFS_freq_stop/gtfs_headway_analysis.py [hunk 1]
```diff
@@ -311,151 +342,173 @@ def get_hlc_stop_frequency(loaded_feeds, start_hour, end_hour, start_date,
...
         crs = GTFS['{}'.format(city)]['crs']
         gtfs_provider = GTFS['{}'.format(city)]['gtfs_provider']
         hour = 'day_time'
-
+        
         # load GTFS Feed
         loaded_feeds = ua_load.gtfsfeed_to_df(gtfsfeed_path=gtfsfeed_path, validation=True, bbox=bbox, remove_stops_outsidebbox=True)
-
+        
+        for root, dirs, files in os.walk(gtfsfeed_path):
```
Comment (shiqin-liu) [inline process/GTFS_freq_stop/gtfs_headway_analysis.py]: So does that mean if the frequency file exists, that would be our preferred option to use for calculating headway? I think it makes sense, as mentioned in tidytransit, it is more accurate than using stop_time alone? The code looks good, just want to make sure that's what we are aiming for :).

File: process/GTFS_freq_stop/gtfs_headway_analysis.py [hunk 2]
```diff
@@ -311,151 +342,173 @@ def get_hlc_stop_frequency(loaded_feeds, start_hour, end_hour, start_date,
...
         end_date = city_config['end_date_mmdd']
         authority = city_config['gtfs_provider']
         bbox = GTFS['{}'.format(city)]['bbox']
         crs = GTFS['{}'.format(city)]['crs']
         gtfs_provider = GTFS['{}'.format(city)]['gtfs_provider']
         hour = 'day_time'
-
+        
         # load GTFS Feed
         loaded_feeds = ua_load.gtfsfeed_to_df(gtfsfeed_path=gtfsfeed_path, validation=True, bbox=bbox, remove_stops_outsidebbox=True)
```
Comment (shiqin-liu) [inline process/GTFS_freq_stop/gtfs_headway_analysis.py]: We need to either update the bbox information in config with new boudaries, or take out the bbox option `ua_load.gtfsfeed_to_df(gtfsfeed_path=gtfsfeed_path)` to load all stops.

**Comment by shiqin-liu** [comment]
Copying your notes from email here:

> - bring in frequencies.txt (urban access doesn't do this it seems)
> - simplified some of the datetime parsing (allowed some things to be written more concisely)
> - added direction_id, and ensured there were no nulls (to allow grouping over this, avoid logic loop - less code)
> - ensure headways are averaged across dates, and then the lowest average headway in either direction is taken 
> - comment out the 'representative week' code -- this code caused Mexico City trains (i think, something like that) not to be processed.  We average over dates anyway, so I think this is fair. 
> - weight hours in frequency.txt  (for taking weighted average of headway for hours within analysis window; hours outside analysis window are weighted zero --- but before then, any zero weight rows are dropped anyway)
> - if there are stops with valid trip frequencies and positive weight defined, these will be used; any other stops are defined using stop_times if possible

Thanks Carl! I think these changes all look good. Sorry that I am not able to test out these on my only laptop as I am running our process code in master branch. But if it works at yours, it should be good! I will wait for your confirmation to merge this PR :).

**Comment by carlhiggs** [comment]
Thanks Shirley --- I'll aim to look at this more later today (maybe we could specify input boundary location in config to automatically derive bbox?); there's an issue with input data for a couple of other cities though, so I'll look at that first.  Hope that's okay, Carl.

**Pull request merged**